core/frontend/src/tile/map/MapLayerAuthentication.ts

```diff
@@ -35,11 +35,26 @@ export interface MapLayerAccessTokenParams {
 }
 
 /** @beta */
-export interface MapLayerAccessClient {
+export interface MapLayerAccessClient  {
   getAccessToken(params: MapLayerAccessTokenParams): Promise<MapLayerAccessToken | undefined>;
   getTokenServiceEndPoint?(mapLayerUrl: string): Promise<MapLayerTokenEndpoint | undefined>;
   invalidateToken?(token: MapLayerAccessToken): boolean;
 
   onOAuthProcessEnd?: BeEvent<Listener>;
 }
 
+/**
+ *  Represents a map layer session manager.
+ * @beta
+ **/
+export interface MapLayerSessionManager {
```

Comment:
Documentation comment merely repeats the interface name. What is this interface for and what does its single property `type` mean?

Reply:
done

Reply:
This type needs to be sub-classed by our consumers, is `type` sufficient ?

Reply:
I don't know because I have no idea what `type` is supposed to mean (it's undocumented). `type` is not even documented. What's it for, what are valid values, why isn't it read-only? Does ever may layer format's session manager actually require a `type` field? Is `MapLayerSessionManager` just some opaque container for name-value pairs specific to the private implementation of a format? If there are answers to all these question please ensure the documentation makes them obvious.

Reply:
@pmconne 
I need an abstraction that will let "external" implementations pass a  Google maps session manager (or any other session manager) down to the provider.

If you have better design suggestions I'm all ears.

Reply:
I don't understand why any of this session manager stuff needs a generic API in core-frontend, vs just letting GoogleMapsImageryProvider (or any other provider) handle it themselves. core-frontend doesn't call `getSessionManager` itself. No one at all calls `setSessionClent`. Only `GoogleMapsImageryProvider.getSessionManager` calls `getSessionClient`, and does a bunch of custom logic with it. No one looks at the `type` field. Why can't the provider just store the client itself?

Reply:
@pmconne I completely removed dependencies to `MapLayerFormatRegistry`.  Please have another look.